src/components/signUpComponent.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'proptypes';
+import logo from '../assets/img/logo.png';
+import loader from '../assets/img/loader.gif';
+
+class SignUpComponent extends Component {
+    state = {
+        username: '',
+        email: '',
+        password: '',
+        confirm_password: '',
+    }
+
+    handleChange = (event) => {
+        this.setState({ [event.target.name]: event.target.value});
+    }
+
+    onSignUp = (event) => {
+        event.preventDefault();
+        const { password, confirm_password } = this.state;
+        if (password === confirm_password) {
+          const { signup } = this.props;
+          signup(this.state);
+        } else {
+          this.setState({ passwordError: 'Passwords don\'t match.' });
+        }
+      }
+
+    render () {
+        const { loading, success, error } = this.props;
+        const { errors } = error ? error.response.data : { errors: '' };
+        const { username, email, password, confirm_password } = this.state;
+        return(
+            <div id="body" className="signup-body">
+                <div className="form-content">
+                    {success ? <Redirect to="/login.html" /> : null}
+                    <div className="logo">
+                        <img src={ logo } alt="Blogger"/>
+                    </div>
+                    <form onSubmit={this.onSignUp}>
+                        <input
```

Comment:
Can be a separate reusable component `<Input/>` that accepts the name, placeholder, default value and on change value"